src/pybind/mgr/dashboard/services/sso.py

```diff
@@ -186,13 +187,13 @@ def handle_sso_command(cmd):
             # pylint: disable=redefined-builtin
             FileNotFoundError = IOError
         try:
-            f = open(sp_x_509_cert, 'r')
+            f = open(sp_x_509_cert, 'r', encoding='utf-8') if six.PY3 else open(sp_x_509_cert, 'rb')
```

Comment:
2 alternatives:
* From [this](https://stackoverflow.com/questions/10971033/backporting-python-3-openencoding-utf-8-to-python-2):
```py
from io import open # py27 - py3 compatible

open(sp_x_509_cert, 'r')  # io.open 'encoding' defaults to 'UTF-8'
>>> <_io.TextIOWrapper name='/tmp/ycm_0lpood4k.log' encoding='utf-8'>
```
* Alternatively, what about a more compact form?
```suggestion
            f = open(sp_x_509_cert, 'r', **{'encoding':'utf-8'} if six.PY3 else {})
```

Reply:
I tried this solution (```from io```) but it does not work for py2. i.e. later idp_metadata parsing crashes.